client/sites-dashboard/components/sites-grid-item.tsx

```diff
@@ -73,6 +74,9 @@ export const SitesGridItem = memo( ( { site }: SitesGridItemProps ) => {
 	const isP2Site = site.options?.is_wpforteams_site;
 	const translatedStatus = useSiteLaunchStatusLabel( site );
 
+	const [ inViewOnce, setInViewOnce ] = useState( false );
+	const ref = useInView< HTMLDivElement >( () => setInViewOnce( true ) );
```

Comment:
Using the `inViewOnce` trick does speed things up in practice, but I'm not really sure why we'd do this over just only rendering it when it was `inView` i.e. removing the DOM elements when they're no longer in view. Since reducing the number of DOM elements is what's giving us the perf boost.